scripts/es2k/setup_infra.sh

```diff
@@ -192,19 +210,44 @@ function run_infrap4d () {
   #gdb --args $P4CP_INSTALL/sbin/infrap4d -grpc_open_insecure_mode=true --nodetach
   #gdb --args $P4CP_INSTALL/sbin/infrap4d --nodetach
   $P4CP_INSTALL/sbin/infrap4d
-  check_status $? "sbin/infrap4d"
+  getPid=$(pgrep -f infrap4d)
+  if [ $getPid ]; then
```

Comment:
It may be worth introducing a small delay between lines 212 and 213, as in some cases infrap4d might take a second or two before it fails.

Reply:
Kernel would right away assign a pid on 212. So there is no need for a time lapse.

Reply:
It's also a daemon process.